tests/source/let_else.rs

```diff
@@ -1,3 +1,127 @@
 fn main() {
-    let Some(1) = Some(1) else { return };
+    // Although this won't compile it still parses so make sure we can format empty else blocks
+    let Some(x) = opt else {};
+
+    // let-else may be formatted on a single line if they are "short" and only contain a single expression
+    let Some(x) = opt else { return };
+
+    let Some(x) = opt else { return; };
+
+    let Some(x) = opt else {
+        // nope
+        return;
+    };
```

Comment:
I think there's a missing case of
```rust
let Some(x) = opt else {
    return
}
```

Also I don't see any examples with long patterns after `let`. Probably good to add variation to the types of patterns in general.

Reply:
In particular, struct patterns, tuples and arrays can raise some unique formatting concerns.

Reply:
> I think there's a missing case of
>
> ```rust
> let Some(x) = opt else {
>     return
> }
> ```

I can add a test case for that!


> Also I don't see any examples with long patterns after let

Now that you bring it up I should probably add one or two to make sure that we wrap the `else block` onto its own line if the pattern is multi-lined.